test/integration/integration.test.ts

```diff
@@ -0,0 +1,678 @@
+      const lock = await space1.locks.acquire(lockId);
+
+      // (Start of assertions_4_1)
+      // ...we observe that `acquire()` returns a lock with `pending` status...
+      context.expect(lock.status).toEqual('pending');
+
+      const eventsTriggeredByAcquiringLock = {
+        // ...and that the two aforementioned listeners get called...
+        space1: {
+          locksUpdate: await promisesForEventsTriggeredByAcquiringLock.space1.locksUpdate,
+        },
+        space2: {
+          locksUpdate: await promisesForEventsTriggeredByAcquiringLock.space2.locksUpdate,
+        },
+      };
+
+      // ...and that the following is true for the events they recieve:
+      for (const event of [
+        eventsTriggeredByAcquiringLock.space1.locksUpdate,
+        eventsTriggeredByAcquiringLock.space2.locksUpdate,
+      ]) {
+        // ...its `id` matches the lock ID passed to `space1.locks.acquire()`...
+        context.expect(event.id).toEqual(lockId);
+        // ...and its `clientId` matches that of `spaces1`...
+        context.expect(event.member.clientId).toEqual(clientId1);
+        // ...and it has no `reason` set...
+        context.expect(event.reason).toBeUndefined();
+        // ...and its `status` is `locked`.
+        context.expect(event.status).toEqual('locked');
+      }
+      // (End of assertions_4_1)
+
+      // (scenario_4_2: try to acquire a lock that’s already held)
+
+      // (Start of setup for assertions_4_2)
+      const promisesForEventsTriggeredByAcquiringHeldLock = {
+        // Next, when we add listeners to `space1.locks` and `space2.locks` for the `update` event using `#once`...
+        space1: {
+          locksUpdate: new Promise<LocksEventMap['update']>((resolve) => {
+            space1.locks.once('update', resolve);
+          }),
+        },
+        space2: {
+          locksUpdate: new Promise<LocksEventMap['update']>((resolve) => {
+            space2.locks.once('update', resolve);
+          }),
+        },
+      };
+      // (End of setup for assertions_4_2)
+
+      // ...and then call `acquire()` on `space2.locks`, passing the same lock ID...
+      const space2Lock = await space2.locks.acquire(lockId);
+
+      // (Start of assertions_4_2)
+      // ...we observe that `acquire()` returns a lock with `pending` status...
+      context.expect(space2Lock.status).toEqual('pending');
+
+      const eventsTriggeredByAcquiringHeldLock = {
+        // ...and that the two aforementioned listeners get called...
+        space1: {
+          locksUpdate: await promisesForEventsTriggeredByAcquiringHeldLock.space1.locksUpdate,
+        },
+        space2: {
+          locksUpdate: await promisesForEventsTriggeredByAcquiringHeldLock.space2.locksUpdate,
+        },
+      };
+
+      // TODO Why is space1 emitting anything here? The emitted events make it appear as though space1 has lost the lock but space2 also failed to acquire it, which seems surprising.
```

Comment:
This is a side-effect of how messages are processed right now; it is not desired but expected right now (there are tickets in the backlog to make that cleaner).

Reply:
Is there an issue that I could add a link to here?

Reply:
Not for this particular case, no - if you could open one, that would be great.

For now, I'd leave a comment here that this behaviour is expected, but not desired.